src/change/sdp.rs

```diff
@@ -193,33 +193,50 @@ impl<'a> SdpApi<'a> {
     /// The mid is not valid to use until the SDP offer-answer dance is complete and
     /// the mid been advertised via [`Event::MediaAdded`][crate::Event::MediaAdded].
     ///
+    /// * `stream_id` is used to synchronize media. It is `a=msid-semantic: WMS <streamId>` line in SDP.
+    /// * `track_id` is becomes both the track id in `a=msid <streamId> <trackId>` as well as the
+    ///   CNAME in the RTP SDES.
+    ///
     /// ```
     /// # use str0m::{Rtc, media::MediaKind, media::Direction};
     /// let mut rtc = Rtc::new();
     ///
     /// let mut changes = rtc.sdp_api();
     ///
-    /// let mid = changes.add_media(MediaKind::Audio, Direction::SendRecv, None);
+    /// let mid = changes.add_media(MediaKind::Audio, Direction::SendRecv, None, None);
     /// ```
-    pub fn add_media(&mut self, kind: MediaKind, dir: Direction, cname: Option<String>) -> Mid {
+    pub fn add_media(
+        &mut self,
+        kind: MediaKind,
+        dir: Direction,
+        stream_id: Option<String>,
+        track_id: Option<String>,
+    ) -> Mid {
         let mid = self.rtc.new_mid();
 
-        let cname = if let Some(cname) = cname {
-            fn is_token_char(c: &char) -> bool {
-                // token-char = %x21 / %x23-27 / %x2A-2B / %x2D-2E / %x30-39
-                // / %x41-5A / %x5E-7E
-                let u = *c as u32;
-                u == 0x21
-                    || (0x23..=0x27).contains(&u)
-                    || (0x2a..=0x2b).contains(&u)
-                    || (0x2d..=0x2e).contains(&u)
-                    || (0x30..=0x39).contains(&u)
-                    || (0x41..=0x5a).contains(&u)
-                    || (0x5e..0x7e).contains(&u)
-            }
-            // https://www.rfc-editor.org/rfc/rfc8830
-            // msid-id = 1*64token-char
-            cname.chars().filter(is_token_char).take(64).collect()
+        // https://www.rfc-editor.org/rfc/rfc8830
+        // msid-id = 1*64token-char
+        fn is_token_char(c: &char) -> bool {
+            // token-char = %x21 / %x23-27 / %x2A-2B / %x2D-2E / %x30-39
+            // / %x41-5A / %x5E-7E
+            let u = *c as u32;
+            u == 0x21
+                || (0x23..=0x27).contains(&u)
+                || (0x2a..=0x2b).contains(&u)
+                || (0x2d..=0x2e).contains(&u)
+                || (0x30..=0x39).contains(&u)
+                || (0x41..=0x5a).contains(&u)
+                || (0x5e..0x7e).contains(&u)
+        }
+
+        let stream_id = if let Some(stream_id) = stream_id {
+            stream_id.chars().filter(is_token_char).take(64).collect()
```

Comment:
Should probably error if the user used an illegal stream id instead of silently fixing it

Reply:
Maybe. Don't want to be overly restrictive.

Reply:
We'll the way it works now it might cause confusing errors instead, fail fast after all ;)

Reply:
I'm more in the camp of providing an easy-to-use API than an overly correct API. Rust libs tend to go for super correct. It's a philosophical question.
 
"What are the chances the user provides some crazy value here?"
Probably not that high. 

"Is it worth introducing yet another Result return type the user has to deal with?"
Maybe, let's see how it goes.

Reply:
Could just panic too, in the spirit of the rest of the lib I think

Reply:
Sure. Panics are nice.